src/models/cart.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable max-len */
+const mongoose = require('mongoose');
+
+const cartSchema = mongoose.Schema({
+  username: { type: String, required: true },
+  // object of dynamic keys. As shown in the sample below, 1 is the productId.
+  // Cannot define dynamic keys in mongo schema
+  products: { type: Object, default: {} },
+}, { minimize: false }); // {1: {sizeSelected: 's', amount: 3 }, {sizeSelected: 'm', amount: 6 },..},
```

Comment:
Oh no, these comments were made by me. I did not copy paste anything here. The comment: 
```// {1: {sizeSelected: 's', amount: 3 }, {sizeSelected: 'm', amount: 6 },..},``` was kinda misplaced. It was meant to describe how the products field within a cart will look like. 

I was just wanted to highlight how the products object will look like by giving an example
```
cart = {
  username: 'antonio@gmail.com`,
  products = {
    productId1: {
      'S': 3
      'M': 4
    },
    productId2: {
      'S': 3
      'M': 4
    },
    productId3: {
      'L': 1
      'M': 1
    },
  }
}
```

I will update the comments to make it look nicer.

Reply:
Ah ok. I understand why you made the comments now. 

However, where possible, you should try to have your code be self-documenting. And, an object model is the perfect case for that. In fact, the object model is documentation for what kind of object a particular model is expecting. My previous comment regarding using an array instead of object literals for the products makes it easier to define a self-documenting object model. So, it would look something like this...

```
const cartSchema = mongoose.Schema({
  username: { type: String, required: true },
  products: [{
    id: { type: String, required: true },
    size: { type: String, enum: ['small', 'medium', 'large'] },
    quantity: { type: Integer, required: true }
  }],
}, { minimize: false });
```